### DISCUSSION TITLE
Rest API Doc

### DISCUSSION THREAD

**Discussion by JShotiko** [minio/minio#15667] [Q&A] [answered]
Is there any documentation of minio rest api? I need requests to get data from minio in json format and use it in other things but all I could find in docs are ways to parse this data with prometheus and I don't want to use prometheus. I need to build simple API parser

**Comment by ravindk89** [comment, marked as answer]
The MinIO S3 REST API is the Amazon S3 REST API. We are API compatible, which is how you can take an application designed for S3 and point it at MinIO without changing much code (beyond hostnames and access credentials). 

The hard part with REST is doing the Signature required by S3. You can look up some guides for this to see whether there's something that works for you.

It's worth noting that our JavaScript SDK returns objects as a `Stream` object. If the original object is JSON, you can probably just parse the stream into JSON yourself.

If the object is some other arbitrary file format, you'd have to be responsible for figuring out how to make that work.

A few more details of your use case might also help here.
